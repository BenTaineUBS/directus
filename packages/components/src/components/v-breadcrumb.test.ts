import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import VBreadcrumb from './v-breadcrumb.vue';
import { h } from 'vue';

const global = {
	components: {
		'v-icon': (props: any) => h('span', { class: 'icon' }, props.name),
		'router-link': (props: any, { slots }: any) => h('a', { href: props.to }, slots),
	},
};

test('Mount component', () => {
	expect(VBreadcrumb).toBeTruthy();

	const wrapper = mount(VBreadcrumb, {
		props: {
			items: [
				{
					to: 'hi',
					name: 'Hi',
				},
				{
					to: 'wow',
					name: 'Wow',
					icon: 'close',
				},
				{
					to: 'disabled',
					name: 'Disabled',
					disabled: true,
				},
			],
		},
		global,
	});

	expect(wrapper.html()).toMatchSnapshot();
});
