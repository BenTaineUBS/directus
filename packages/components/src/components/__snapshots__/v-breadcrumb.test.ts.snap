// Vitest Snapshot v1

exports[`Mount component 1`] = `"<span class=\\"v-breadcrumb\\" data-v-61eab8ca=\\"\\"><span class=\\"section\\" data-v-61eab8ca=\\"\\"><!--v-if--><a href=\\"hi\\" class=\\"section-link\\" data-v-61eab8ca=\\"\\"><!--v-if--> Hi</a></span><span class=\\"section\\" data-v-61eab8ca=\\"\\"><span class=\\"icon\\" data-v-61eab8ca=\\"\\">chevron_right</span><a href=\\"wow\\" class=\\"section-link\\" data-v-61eab8ca=\\"\\"><span class=\\"icon\\" data-v-61eab8ca=\\"\\">close</span> Wow</a></span><span class=\\"section disabled\\" data-v-61eab8ca=\\"\\"><span class=\\"icon\\" data-v-61eab8ca=\\"\\">chevron_right</span><span class=\\"section-link\\" data-v-61eab8ca=\\"\\"><!--v-if--> Disabled</span></span></span>"`;
